tactical-microgrid-standard/common/QosHelper.cpp

```diff
@@ -257,67 +251,107 @@ DDS::SubscriberQos get_qos()
 
 namespace DataReader {
 
-DDS::DataReaderQos get_PublishLast(const tms::Identity& device_id)
+const DDS::DataReaderQos& get_PublishLast(const tms::Identity& device_id)
 {
-  DDS::DataReaderQos qos;
-  init_endpoint_PublishLast_profile(qos, device_id);
-  init_datareader_common(qos);
+  static DDS::DataReaderQos qos;
+  static bool ready = false;
+  if (!ready) {
+    init_endpoint_PublishLast_profile(qos);
+    init_datareader_common(qos);
+    ready = true;
+  }
```

Comment:
Instead of the explicit bool, we can let the compiler manage initialization for us:

```
  static DDS::DataReaderQos qos = init(); // combined effects of the 2 init_* functions currently called in this scope
```

This could also (probably?) be an immediately-invoked lambda to avoid needing to define other functions


```
  static DDS::DataReaderQos qos = []() {
    DDS::DataReaderQos qos;
    // combined effects of the 2 init_* functions currently called in this scope
    return qos;
  }();
```